game/theater/iadsnetwork/iadsnetwork.py

```diff
@@ -112,6 +112,17 @@ def __init__(
             else:
                 raise RuntimeError("Invalid iads_config in campaign")
 
+    @property
+    def participating(self) -> Iterator[TheaterGroundObject]:
+        """All unique participating TGOs. First primary then secondary"""
+        secondary_nodes = []
+        for node in self.nodes:
+            yield node.group.ground_object
+            for connection in node.connections.values():
+                if connection.ground_object not in secondary_nodes:
```

Comment:
Ofc i can do that :)
It is because this function iterates ovet all connections betwee primary nodes (sam, ewr, cc) and secondary nodes (comm or power). While each node is unique the connections are not. One secondary node can have 1..N connections to primary nodes. Therefore the check for dups.

Tbh im thinking about that function if it really is such a benefit to check the dups rather then just live with dups. Performance wise it could be equal.